FTBFS with GCC 13

### Godot version

4.0

### System information

Linux

### Issue description

Please cherry-pick https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/commit/29d492b60c84ca784ea0943efc7d2e6e0f3bdaac in order to support the upcoming GCC 13 compiler.

### Steps to reproduce

Build with gcc 13.

### Minimal reproduction project

...

The build failure is:
```
[  263s] In file included from thirdparty/vulkan/vk_mem_alloc.cpp:7:
[  263s] thirdparty/vulkan/vk_mem_alloc.h: In function 'void VmaUint32ToStr(char*, size_t, uint32_t)':
[  263s] thirdparty/vulkan/vk_mem_alloc.h:2821:9: error: 'snprintf' was not declared in this scope
[  263s]  2821 |         snprintf(outStr, strLen, "%u", static_cast<unsigned int>(num));
[  263s]       |         ^~~~~~~~
[  263s] thirdparty/vulkan/vk_mem_alloc.h:2655:1: note: 'snprintf' is defined in header '<cstdio>'; did you forget to '#include <cstdio>'?
[  263s]  2654 |     #include <mutex>
[  263s]   +++ |+#include <cstdio>
[  263s]  2655 | #else
```

--- a/thirdparty/vulkan/patches/VMA-fix-gcc13.patch
+++ b/thirdparty/vulkan/patches/VMA-fix-gcc13.patch
@@ -1,0 +1,26 @@
+From 29d492b60c84ca784ea0943efc7d2e6e0f3bdaac Mon Sep 17 00:00:00 2001
+From: Adam Sawicki <adam.sawicki@amd.com>
+Date: Thu, 19 Jan 2023 13:19:55 +0100
+Subject: [PATCH] Added missing #include <cstdio>
+
+For snprintf, for compatibility with GCC 13.
+Fixes #312 - thanks @marxin !
+---
+ thirdparty/vulkan/vk_mem_alloc.h | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/thirdparty/vulkan/vk_mem_alloc.h b/thirdparty/vulkan/vk_mem_alloc.h
+index b787c36..0fe459b 100644
+--- a/thirdparty/vulkan/vk_mem_alloc.h
++++ b/thirdparty/vulkan/vk_mem_alloc.h
+@@ -2614,6 +2614,10 @@ VMA_CALL_PRE void VMA_CALL_POST vmaFreeStatsString(
+     #include <bit> // For std::popcount
+ #endif
+ 
++#if VMA_STATS_STRING_ENABLED
++    #include <cstdio> // For snprintf
++#endif
++
+ /*******************************************************************************
+ CONFIGURATION SECTION
+ 

--- a/thirdparty/vulkan/vk_mem_alloc.h
+++ b/thirdparty/vulkan/vk_mem_alloc.h
@@ -2582,6 +2582,10 @@ VMA_CALL_PRE void VMA_CALL_POST vmaFreeStatsString(
     #include <bit> // For std::popcount
 #endif
 
+#if VMA_STATS_STRING_ENABLED
+    #include <cstdio> // For snprintf
+#endif
+
 /*******************************************************************************
 CONFIGURATION SECTION
 

